src/ControlSystem/ControlErrors/Size/ComovingCharSpeedDerivative.hpp

```diff
@@ -0,0 +1,347 @@
+ * is the distorted-frame
+ * component of the grid-frame shift, defined by
+ * \f{align}
+ * \beta^\hat{i} &= \beta^i \frac{\partial \hat{x}^\hat{i}}{\partial x^i}.
+ * \label{eq:shiftyquantity}
+ * \f}
+ * This is **not** the shift in the distorted frame, because the shift does
+ * not transform like a spatial tensor under the maps.
+ *
+ * If the comoving characteristic speed \f$v_c\f$ is negative and remains
+ * negative forever, then size control will fail.  Therefore \f$v_c\f$ is
+ * used in making decisions in size control. We care about
+ * \f$d v_c/d\lambda_{00}\f$ because the sign of that quantity tells us
+ * whether \f$v_c\f$ will increase or decrease when we increase or decrease
+ * the map parameter \f$\lambda_{00}\f$; this information will be used to
+ * decide whether to transition between different size control states.
+ *
+ * ## Derivation of comoving_char_speed_derivative
+ *
+ * The function comoving_char_speed_derivative computes
+ * \f$d v_c/d\lambda_{00}\f$ on the excision boundary, where the total
+ * derivative means that all other map parameters
+ * (like \f$\lambda_{\ell m}\f$ for \f$\ell>0\f$) are held fixed, and the
+ * coordinates of the excision boundary (the grid coordinates) are held fixed.
+ * We also hold fixed \f$\dot{\lambda}_{00}\f$ because we are interested
+ * in how $v_c$ changes from a configuration with a given
+ * \f$\lambda_{00}\f$ and \f$\dot{\lambda}_{00}\sim 0\f$ to another
+ * configuration with a different nearby \f$\lambda_{00}\f$ and also with
+ * \f$\dot{\lambda}_{00}\sim 0\f$.
+ *
+ * Here we derive an expression for \f$d v_c/d\lambda_{00}\f$.
+ * This expression will be
+ * complicated, mostly because of the normals \f$\hat{n}_\hat{i}\f$ that appear
+ * in Eq. (\f$\ref{eq:comovingspeed}\f$) and because of the Jacobians.
+ *
+ * ### Derivative of the Jacobian
+ *
+ * First, note that by differentiating Eq. (\f$\ref{eq:map}\f$) we obtain
+ * \f{align}
+ *   \frac{d\hat{x}^i}{d\lambda_{00}}
+ *      &= - \frac{x^i Y_{00}}{r_{\mathrm{EB}}} \\
+ *      &= -\xi^i Y_{00},
+ * \f}
+ * where the last line follows from the definition of the direction cosines.
+ *
+ * The Jacobian of the map is
+ * \f{align}
+ *  \frac{\partial \hat{x}^i}{\partial x^j}
+ *    &= (1 - \lambda_{00} Y_{00}/r_{\mathrm{EB}} + B) \delta^i_j
+ *    + x^i \frac{\partial B}{\partial x^j},
+ * \f}
+ * where the last term is independent of \f$\lambda_{00}\f$.
+ * Therefore, we have
+ * \f{align}
+ *  \frac{d}{d\lambda_{00}}\frac{\partial \hat{x}^i}{\partial x^j} &=
+ *  -\frac{Y_{00}}{r_{\mathrm{EB}}} \delta^i_j.
+ * \f}
+ *
+ * By taking the derivative of the identity
+ * \f{align}
+ *  \frac{\partial \hat{x}^\hat{i}}{\partial x^k}
+ *  \frac{\partial x^k}{\partial \hat{x}^\hat{j}} &= \delta^\hat{i}_\hat{j}
+ * \f}
+ * we can derive
+ * \f{align}
+ *  \frac{d}{d\lambda_{00}}\frac{\partial x^i}{\partial \hat{x}^j} &=
+ *  +\frac{Y_{00}}{r_{\mathrm{EB}}} \delta^i_j.
+ * \f}
```

Comment:
Uh oh. When I redid the calculation more carefully (making sure all the hats are in the right place), I still get the sign change but I get an extra factor of Jacobian.  This will make all the subsequent formulas more complicated, and may mean that yet another quantity might need to be interpolated.

Reply:
Hmm yes I can see that...Is the possible new quantity the jacobian itself?

Reply:
Now pass in a Jacobian.